discord/commands/mixins.py

```diff
@@ -0,0 +1,952 @@
+        return self._callback
+
+    @callback.setter
+    def callback(self, func: Callback) -> None:
+        if not asyncio.iscoroutinefunction(func):
+            raise TypeError("Callback must be a coroutine.")
+
+        self._callback = func
+        unwrap = unwrap_function(func)
+        self.module = unwrap.__module__
+
+    @property
+    def cooldown(self) -> Cooldown | None:
+        """Returns the cooldown for the command."""
+        return self._buckets._cooldown
+
+    @property
+    def qualified_name(self) -> str:
+        """Retrieves the fully qualified command name.
+
+        This is the full name of the parent command with the subcommand name as well.
+        For example, in ``?one two three``, the qualified name would be
+        ``one two three``.
+        """
+        if not self.parent:
+            return self.name
+
+        return f"{self.parent.qualified_name} {self.name}"
+
+    @property
+    def cog_name(self) -> str | None:
+        """Optional[:class:`str`]: The name of the cog this command belongs to, if any."""
+        return type(self.cog).__cog_name__ if self.cog is not None else None
+
+    @property
+    def parents(self) -> list[Invokable]:
+        """List[:class:`Invokable`]: Retrieves the parents of this command.
+
+        If the command has no parents then it returns an empty :class:`list`.
+
+        For example in commands ``?a b c test``, the parents are ``[c, b, a]``.
+        """
+        entries = []
+        command = self
+        while command.parent is not None:
+            command = command.parent
+            entries.append(command)
+
+        return entries
+
+    @property
+    def root_parent(self) -> Invokable | None:
+        """Optional[:class:`Invokable`]: Retrieves the root parent of this command.
+
+        If the command has no parents then it returns ``None``.
+
+        For example in commands ``?a b c test``, the root parent is ``a``.
+        """
+        if not self.parent:
+            return None
+        return self.parents[-1]
+
+    @property
+    def full_parent_name(self) -> str | None:
+        """Retrieves the fully qualified parent command name.
+
+        This the base command name required to execute it. For example,
+        in ``/one two three`` the parent name would be ``one two``.
+        """
+        if self.parent:
+            return self.parent.qualified_name
+
+    def __str__(self) -> str:
+        return self.qualified_name
+
+    async def __call__(self, ctx: BaseContext, *args: P.args, **kwargs: P.kwargs):
+        """|coro|
+
+        Calls the internal callback that the command holds.
+
+        .. note::
+
+            This bypasses all mechanisms -- including checks, converters,
+            invoke hooks, cooldowns, etc. You must take care to pass
+            the proper arguments and types to this function.
+        """
+        new_args = (ctx, *args)
+        if self.cog is not None:
+            new_args = (self.cog, *args)
```

Comment:
```suggestion
            new_args = (self.cog, *new_args)
```
Otherwise the added `ctx` argument would be lost for commands in cogs.